### PR TITLE
[MM-28256] Cypress/E2E: Automate backlogs - Slash Commands > Statuses (4 test cases)

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary

E2E testing for https://mattermost.atlassian.net/browse/MM-28256

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28256

### CODE REVIEW

File: e2e/cypress/integration/menus/status_dropdown_spec.js [hunk 1]
```diff
@@ -36,6 +48,106 @@ describe('Status dropdown menu', () => {
...
+
+        // # On modal that asks if you want to be set Online, select Yes.
+        cy.get('.modal-content').within(() => {
+            cy.findByText('Your Status is Set to "Away"').should('be.visible');
+            cy.findByText('Yes, set my status to "Online"').click();
+        });
+
+        // * Your status is online in your view
+        cy.get('.online--icon').should('be.visible');
+
```
Comment (jfrerich) [inline e2e/cypress/integration/menus/status_dropdown_spec.js:77]: @saturninoabril, if I place a cy.pause() at this point, I can view see who user1 is and open a DM with that user as a test user in MM. After some time, the user status shows them as online. However, after I login through the cypress test below this line and open a DM with user1, user1 never shows up as online. Thus, this test fails after this point. I'm not sure how to simulate this in cypress. I tried adding a minute of waite time after opening the DM, but user1 never shows up as online

Comment (saturninoabril) [reply e2e/cypress/integration/menus/status_dropdown_spec.js:77]: I'm seeing that too. I tried several approach and it's not working either. Will take a look deeper after the Hackathon.  Thanks for checking!

Comment (saturninoabril) [reply e2e/cypress/integration/menus/status_dropdown_spec.js:77]: I got it working by adding extra check to status after click. Please try below and let me know if will work on you as well.
```javascript
cy.get('.modal-content').within(() => {
    cy.findByText('Your Status is Set to "Away"').should('be.visible');
    cy.findByText('Yes, set my status to "Online"').click();

    cy.apiGetUserStatus(user1.id).its('status.status').should('equal', 'online');
});

// * Your status is online in your view
cy.get('.online--icon').should('be.visible');
```

File: e2e/cypress/integration/menus/status_dropdown_spec.js [hunk 2]
```diff
@@ -36,6 +48,106 @@ describe('Status dropdown menu', () => {
...
+        // # login in as user2
+        cy.apiLogin(user2);
+        cy.visit(testChannelUrl);
+        openDM(user1.username);
+
+        // * Your status is online in other users' views.
+        cy.apiGetUserStatus(`${user1.id}`).then((result) => {
+            cy.wrap(result.status.status).should('be.equal', 'online');
+        });
```
Comment (saturninoabril) [inline e2e/cypress/integration/menus/status_dropdown_spec.js:86]: Unfortunately, I couldn't get this working without forcing `user1` to update its status to online externally
```javascript
cy.apiLogin(user2);
cy.visit(testChannelUrl);
cy.makeClient({user: user1}).then((client) => {
    cy.wrap(client.updateStatus({
        user_id: user1.id,
        status: 'online',
        manual: true,
    })).then((response) => {
        expect(response.status).to.equal('online');
        openDM(user1.username);

        // * Your status is online in other users' views.
        cy.apiGetUserStatus(user1.id).its('status.status').should('equal', 'online');
    });
});
```
Note: There's a fix for `cy.makeClient` if you'd like to try above - https://github.com/mattermost/mattermost-webapp/pull/7227/files

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Hello @jfrerich, just a general comment before making detailed review (once you make an update to this PR). Could you please move this new test into its own spec file? I see some failing test on existing tests but couldn't see immediately if it's due to the change in this PR.

File: e2e/cypress/integration/menus/status_dropdown_spec.js [hunk 3]
```diff
@@ -10,11 +10,23 @@
 // Stage: @prod
 // Group: @menu
```
Comment (prapti) [inline e2e/cypress/integration/menus/status_dropdown_spec.js:11]:
```suggestion
// Group: @menu
```

**Review by prapti — APPROVED** [review]
Other than the outstanding issues, the rest looks good to me. The only suggestion I have is to remove the `Stage` since there are a few new tests going in this file. We can bump this back to production once the new tests and changes 'soak in' for a few runs.
Thank you so much @jfrerich!

**Comment by DHaussermann** [comment]
@jfrerich is there any further change needed here to address the changes mentioned by Saturn above?

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to https://github.com/mattermost/mattermost-server/issues/22420 for status updates on this effort.

**Pull request closed without merging**